marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs

```diff
@@ -328,6 +352,114 @@ class FlowUtils {
+      }
+      let childNodes = content.childNodes;
+      for (let i = 0; i < childNodes.length; i++) {
+        let childNode = childNodes[i];
+        if (childNode instanceof Element) {
+          organizedOutput[childNode.localName] = organizedOutput[childNode.localName] || [];
+          organizedOutput[childNode.localName].push(this.xmlNodeToJson(childNode));
+        } else {
+          organizedOutput['$text'] = organizedOutput['$text'] || [];
+          organizedOutput['$text'].push(this.xmlNodeToJson(childNode));
+        }
+      }
+      if (organizedOutput['$text'] && fn.normalizeSpace(organizedOutput['$text'].join('')) === '') {
+        delete organizedOutput['$text'];
+      }
+      for (let key in organizedOutput) {
+        if (organizedOutput.hasOwnProperty(key) && organizedOutput[key].length === 1) {
+          organizedOutput[key] = organizedOutput[key][0];
+        }
+      }
+      if (Object.keys(organizedOutput).length === 1 && organizedOutput['$text']) {
+        return organizedOutput['$text'];
+      } else {
+        return organizedOutput;
+      }
+    } else {
+      return fn.string(content);
+    }
+  }
+
+  jsonToXml(content) {
+    let contentInput = content;
+    if (content instanceof ObjectNode || content instanceof ArrayNode) {
+      contentInput = content.toObject();
+    }
+    return this.jsonToXmlNodeBuilder(contentInput).toNode();
+  }
+
+  jsonToXmlNodeBuilder(content, nb = new NodeBuilder()) {
+    if (content instanceof Object) {
+      for (let propName in content) {
+        if (content.hasOwnProperty(propName)) {
+          let propValues = content[propName];
+          if (propValues instanceof Array) {
+            for (let propValueIndex in propValues) {
+              if (propValues.hasOwnProperty(propValueIndex)) {
+                nb.startElement(propName);
+                this.jsonToXml(propValues[propValueIndex], nb);
+                nb.endElement();
+              }
+            }
+          } else {
+            nb.startElement(propName);
+            this.jsonToXml(propValues, nb);
+            nb.endElement();
+          }
+        }
+      }
+    } else {
+      nb.addText(content);
+    }
+    return nb;
+  }
+
+  evalSubstituteVal(value) {
+    let obj = consts.PROPERY_KEY_MAP.get(value);
+    if (obj === undefined) return value;
+    if (obj == consts.CURRENT_DATE_TIME) {
+      return fn.currentDateTime();
+    } else if (obj == consts.CURRENT_USER) {
+      return xdmp.getCurrentUser();
+    }
+    return value;
+  }
+
+  createHeaders(options) {
+    let headers = {};
+    for (let key in options.headers) {
+      headers[key] = this.evalSubstituteVal(options.headers[key]);
+    }
+    return headers;
+  }
+
+  createMetadata(metaData = {}, flowName, stepName) {
+    metaData[this.consts.CREATED_ON] = this.datahub.flowUtils.evalSubstituteVal(this.consts.CREATED_ON);
+    metaData[this.consts.CREATED_BY] = this.datahubflowUtils.evalSubstituteVal(this.consts.CREATED_BY);
```

Comment:
https://github.com/marklogic/marklogic-data-hub/blob/da4e0f04fbda0b01cb3998274374d41bb9b2297b/marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/flow-utils.sjs#L456-L457

metaData[this.consts.CREATED_ON] = this.evalSubstituteVal(this.consts.CREATED_ON);
metaData[this.consts.CREATED_BY] = this.evalSubstituteVal(this.consts.CREATED_BY);

Reply:
Done.